### PR TITLE
Feat(pair_stable, pair_stable_bluna): remove condition to LP all assets

### PR DESCRIPTION
**Pull request by P-Yevhenii**

### CODE REVIEW

File: contracts/pair_stable/src/contract.rs [hunk 1]
```diff
@@ -1207,8 +1213,10 @@ fn compute_swap(
     );
 
     // We assume the assets should stay in a 1:1 ratio, so the true exchange rate is 1. So any exchange rate <1 could be considered the spread
-    let spread_amount = offer_amount.saturating_sub(return_amount);
-
+    let mut spread_amount = offer_amount.saturating_sub(return_amount);
+    if spread_amount.to_string().len() <= greater_precision as usize {
+        spread_amount = Uint128::zero();
+    }
```
Comment (ShadoySV) [inline contracts/pair_stable/src/contract.rs:1219]: It's not right to drop spread less than 1

File: contracts/pair_stable/src/contract.rs [hunk 2]
```diff
@@ -1344,6 +1352,9 @@ pub fn assert_max_spread(
             return Err(ContractError::MaxSpreadAssertion {});
         }
     } else if Decimal::from_ratio(spread_amount, return_amount + spread_amount) > max_spread {
+        let val = Decimal::from_ratio(spread_amount, return_amount + spread_amount);
+        let val1 = max_spread;
+        let val2 = val - val1;
```
Comment (ShadoySV) [inline contracts/pair_stable/src/contract.rs:1357]: We don't need this

File: contracts/pair_stable/tests/integration.rs [hunk 3]
```diff
@@ -291,6 +291,317 @@ fn provide_liquidity_msg(
...
+            to: None,
+        })
+        .unwrap(),
+        amount: swap_amount,
+    };
+
+    app.execute_contract(owner.clone(), token_x_instance.clone(), &msg, &[])
+        .unwrap();
+
+    // try to provide for single token and increase the ratio in the pool from 1 to 2
```
Comment (ShadoySV) [inline contracts/pair_stable/tests/integration.rs:548]: ... from 1 to 2.5

File: contracts/pair_stable/src/contract.rs [hunk 4]
```diff
@@ -373,27 +373,35 @@ pub fn provide_liquidity(
...
-                    amount: deposits[i],
-                })?,
-                funds: vec![],
-            }))
-        } else {
-            // If the asset is a native token, the pool balance already increased
-            // To calculate the pool balance properly, we should subtract the user deposit from the recorded pool token amount
-            pool.amount = pool.amount.checked_sub(deposits[i])?;
+        // we cannot put a zero amount into an empty pool.
+        if deposits[i].is_zero() && pool.amount.is_zero() {
```
[the remark this answers is not in the document]
Comment (epanchee) [reply contracts/pair_stable/src/contract.rs:383]: Probably it's better to raise an error before this for-loop if both sides of pool are empty and user tries to insert only one asset.

[the remark this answers is not in the document]
Comment (epanchee) [reply contracts/pair_stable/src/contract.rs:383]: Hm, I think your approach is ok. Nevermind, I resolve this comment.

**Review by maksk7 — APPROVED** [review]

**Review by ShadoySV — APPROVED** [review]

**Review by epanchee — APPROVED** [review]

**Pull request merged**